### PR TITLE
add unit testing and replace goutte  with guzzle

### PR DESCRIPTION
**Pull request by peter279k**
I reopen the pull request again.

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,27 +1,26 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<phpunit backupGlobals="false"
-         backupStaticAttributes="false"
-         colors="true"
-         convertErrorsToExceptions="true"
-         convertNoticesToExceptions="true"
-         convertWarningsToExceptions="true"
-         processIsolation="false"
```
Comment (Th3Mouk) [inline phpunit.xml.dist]: Could you add this line please ?

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
-         syntaxCheck="false">
+<phpunit
+    colors="true"
+    bootstrap="./vendor/autoload.php"
+    type="coverage-text"
+    target="tests/"
+    charset="UTF-8">
+
     <testsuites>
-        <testsuite name="Test Suite">
```
Comment (Th3Mouk) [inline phpunit.xml.dist]: Revert please the naming

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
-            <directory >./tests/</directory>
```
Comment (Th3Mouk) [inline phpunit.xml.dist]: Revert the path please

File: phpunit.xml.dist [hunk 4]
```diff
@@ -1,27 +1,26 @@
...
+        <testsuite name="testing">
+      	    <directory suffix=".php">tests/</directory>
         </testsuite>
     </testsuites>
 
     <filter>
         <whitelist>
-            <directory suffix=".php">./src</directory>
```
Comment (Th3Mouk) [inline phpunit.xml.dist]: Revert the path please

**Comment by Th3Mouk** [comment]
Please commit here the changes,  and squash commit after.

**Comment by Th3Mouk** [comment]
Move to #6 😕

**Comment by peter279k** [comment]
Unfortunately, I cannot synchronize this PR because I have this [problem](https://github.com/isaacs/github/issues/168) now.

**Comment by Th3Mouk** [comment]
Understood

**Pull request closed without merging**